### PR TITLE
feat: [QUAD-5555] - adds hardhat support

### PR DESCRIPTION
**Pull request by adam-hanna**
## Motivation
Deploying locally to hardhat is helpful for developers

*JIRA*: NA

## Implementation
Adds support for a local deployment to hardhat.

## Checklist
- [ ] Unit test implemented 
- [ ] Manual test via `local` or `dev` environment
- [ ] Update technical specifications
- [ ] PR title format follows the guide in the `.github/CONTRIBUTING.md`

### CODE REVIEW

File: .openzeppelin/unknown-31337.json [hunk 1]
```diff
@@ -6969,8 +6984,8 @@
       }
     },
     "b1317100b2a90f9b40592600dafb72fdbeba66b4261228ef79990a671d816cd3": {
-      "address": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
```
Comment (adam-hanna) [inline .openzeppelin/unknown-31337.json:6972]: Does changing these cause issues for deploying to int or mainnet?

File: .openzeppelin/unknown-31337.json [hunk 2]
```diff
@@ -7241,8 +7256,8 @@
       }
     },
     "439d93cfaa149ec48a881ea8df4eacea5789ac5972ec3259309ff5626c909369": {
-      "address": "0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9",
```
Comment (adam-hanna) [inline .openzeppelin/unknown-31337.json:7244]: Same comment as above.

File: .openzeppelin/unknown-31337.json (hunk 1 above)
Comment (faze-quadrata) [reply .openzeppelin/unknown-31337.json:6972]: no, should be fine

File: .openzeppelin/unknown-31337.json (hunk 2 above)
Comment (faze-quadrata) [reply .openzeppelin/unknown-31337.json:7244]: no, should be fine, the file says 31337 which is the hardhat chain id

**Review by faze-quadrata — APPROVED** [review]

**Pull request merged**